### DISCUSSION TITLE
How to redirect to index page on the edit page after record deleted

### DISCUSSION THREAD

**Discussion by xiaomlove** [filamentphp/filament#6212] [Help] [answered]
In the edit page, I added a DeleteAction, how can I do something else after deleting the record like clearing the cache and then display the notification normally before jumping to the list page? I tried to write it like this:
```
Actions\DeleteAction::make()->using(function ($record) {
    $record->delete();
    clear_some_cache();
})->successRedirectUrl(SomeResource::getUrl())
```
but it reports the following error:
```
Missing required parameter for [Route: filament.resources.xxx.edit] [URI: xxx/{record}/edit] [Missing parameter: record]
```
I try do the redirect after clear cache function, no error but there is no notification any more, not I want.

So how do I write it so that after deleting a record, it displays the message normally and then jumps to the list page

**Comment by danharrin** [comment, marked as answer]
Its not possible to delay the redirect and show the notification before

**Reply by danharrin** [reply]
Because once the record is deleted, the page can't be re-rendered to show the notification, since the data doesn't exist anymore.

**Reply by xiaomlove** [reply]
So I can only jump directly in the using() callback, I can't show the hint message anymore. Thanks for the answer!
